dali/operators/audio/resample.cc

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <vector>
+#include "dali/operators/audio/resample.h"
+#include "dali/operators/audio/resampling_params.h"
+#include "dali/kernels/kernel_params.h"
+
+namespace dali {
+
+DALI_SCHEMA(experimental__AudioResample)
+  .DocStr(R"(Resamples a signal with a different sampling rate.
```

Comment:
```suggestion
  .DocStr(R"(Resamples a signal with a given sampling rate.
```

Reply:
Actually, neither is true. "given" suggests an absolute value and that's not how this operator works - you don't even need to supply sampling rates directly (just the ratio). Also, I could add a fast path for scale==1 (so if the rate isn't really different, there would be no perf hit).

Reply:
Up to you. I just wanted to point out that not necessarily the sampling rate need to be different.